tests/system-tests/rdscore/internal/rdscorecommon/metallb-traffic-segregation.go

```diff
@@ -0,0 +1,731 @@
+
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Verifying that the correct route was used "+
+			"for the second FRR defined %s/%s", RDSCoreConfig.MetalLBFRRTwoIPv4,
+			RDSCoreConfig.MetalLBFRRTwoIPv6)
+
+		for _, targetHostIP := range []string{RDSCoreConfig.MetalLBTrafficSegregationTargetTwoIPv4,
+			RDSCoreConfig.MetalLBTrafficSegregationTargetTwoIPv6} {
+			if targetHostIP == "" {
+				glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Empty target IP continue")
+
+				continue
+			}
+
+			myIP, err := netip.ParseAddr(targetHostIP)
+			Expect(err).ToNot(HaveOccurred(),
+				fmt.Sprintf("Failed to parse host ip %s", targetHostIP))
+
+			searchString := RDSCoreConfig.MetalLBFRRTwoIPv4
+
+			if myIP.Is6() {
+				searchString = RDSCoreConfig.MetalLBFRRTwoIPv6
+			}
+
+			err = supporttools.SendTrafficFindExpectedString(
+				_pod,
+				targetHostIP,
+				RDSCoreConfig.MetalLBTrafficSegregationTargetPort,
+				searchString)
+			Expect(err).ToNot(HaveOccurred(),
+				fmt.Sprintf("Failed to find required FRR IP address %s in response: %v", searchString, err))
+		}
+	}
+}
+
+// VerifyMetallbIngressTrafficSegregation test ingress connectivity to the workloads running on the cluster exposed
+// via service(of type loadbalancer) with traffic segregation.
+//
+//nolint:funlen
+func VerifyMetallbIngressTrafficSegregation(ctx SpecContext) {
+	By("Asserting if test URLs are provided")
+
+	var err error
+
+	if RDSCoreConfig.MetalLBFRROneIPv4 == "" && RDSCoreConfig.MetalLBFRROneIPv6 == "" {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"First MetalLB FRR IP not specified or are empty. Skipping...")
+		Skip("First MetalLB FRR IP not specified or are empty")
+	}
+
+	if RDSCoreConfig.MetalLBFRRTwoIPv4 == "" && RDSCoreConfig.MetalLBFRRTwoIPv6 == "" {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"Second MetalLB FRR IP not specified or are empty. Skipping...")
+		Skip("Second MetalLB FRR IP not specified or are empty")
+	}
+
+	if RDSCoreConfig.MetalLBTrafficSegregationTCPDumpIntOne == "" &&
+		RDSCoreConfig.MetalLBTrafficSegregationTCPDumpIntTwo == "" {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"One of the capture interfaces not specified or are empty. Skipping...")
+		Skip("One of the capture interfaces not specified or are empty")
+	}
+
+	By("Ensure LB application deployment created for the first bgp route")
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Searching for pods in %q namespace",
+		RDSCoreConfig.MetalLBLoadBalancerOneNamespace)
+
+	lbOnePodsList := findPodWithSelector(RDSCoreConfig.MetalLBLoadBalancerOneNamespace, lbDeploymentLabel)
+
+	var lbOneNodeName string
+
+	if len(lbOnePodsList) == 0 {
+		glog.V(rdscoreparams.RDSCoreLogLevel).Infof(
+			"FRR One Load Balance app not found deployed. Skipping...")
+		Skip("FRR One Load Balance app not found deployed")
+	}
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Getting node name for the loadbalancer one application")
+
+	lbOneNodeName = lbOnePodsList[0].Object.Spec.NodeName
```

Comment:
if application expects only 1 pod - I suggest adding extra check that there's exactly one pod running

Reply:
not sure we need it, the only requirement to have at least one. But, maybe to avoid leftovers from the previous runs it make sense. Was added with the skip condition